README.md

```diff
@@ -198,3 +198,130 @@ L'applicazione utilizza un custom handler per intercettare e notificare errori j
+Per ridefinire il tema dei componenti di native-base è necesario creare/modificare i file presenti nella directory `/js/theme/components`. Ogni file presente in questa directory deve esportare un oggetto che definisce il tema del componente. Prendiamo come esempio il file `Content.js`:
+
+```javascript
+import { type Theme } from '../types'
+import variables from '../variables'
+
+export default (): Theme => {
+  const theme = {
+    padding: variables.contentPadding,
+    backgroundColor: variables.contentBackground
+  }
+
+  return theme
+}
+```
+
+In questo file è possibile notare come vengono ridefiniti due attributi (`padding` e `backgroundColor`) utilizzando come valori quanto presente nelle relative variabili. L'oggetto restituito sarà utilizzato nel file `/js/theme/index` per associarlo ad uno specifico tipo di componente (in questo caso `NativeBase.Component`).
+
+Un esempio più complesso permette di utilizzare le funzioni avanzate del layer di theming di native-base.
+
+```javascript
+import { type Theme } from '../types'
+import variables from '../variables'
+
+export default (): Theme => {
+  const theme = {
+    '.spacer': {
+      '.large': {
+        height: variables.spacerLargeHeight
+      },
+
+      height: variables.spacerHeight
+    },
+
+    '.footer': {
+      paddingTop: variables.footerPaddingTop,
+      paddingLeft: variables.footerPaddingLeft,
+      paddingBottom: variables.footerPaddingBottom,
+      paddingRight: variables.footerPaddingRight,
+      backgroundColor: variables.footerBackground,
+      borderTopWidth: variables.footerShadowWidth,
+      borderColor: variables.footerShadowColor
+    }
+  }
+
+  return theme
+}
+```
+
+All'interno del file del tema di un singolo componente è possibile infatti definire degli attributi specifici che verranno utilizzati sono nel caso il componente in questione abbia una specifica prop.
+Definendo nell'oggetto di tema qualcosa come
+```javascript
+'.footer': {
+      paddingTop: variables.footerPaddingTop
+    }
+```
+sarà possibile quando necessario utilizzare il componente associandogli la property `footer` nel seguente modo `<Component footer />` ed automaticamente il sistema di theming applicherà al componente gli attributi definiti (`paddingTop: variables.footerPaddingTop`).
+
+Altra funzione avanzata è quella che permette di definire il tema dei componenti figli a partire dal componente padre.
+Prediamo come esempio il seguente frammento di codice di un generico componente:
+
+```javascript
+...
+render() {
+  return(
+    <Content>
+      <Button>
+        <Text>My button</Text>
+      </Button>
+    </Content>
+  )
+}
+...
+```
+
+La libreria native-base permette di definire l'aspetto del componente figlio `Text` presente all'interno del componente padre `Button`. Ad esempio per definire la dimensione del testo all'interno di tutti i bottoni presenti nell'applicazione è possibile all'interno del file `/js/theme/components/Button.js` il seguente codice:
```

Comment:
Ad esempio per definire la dimensione del testo in tutti i bottoni presenti nell'applicazione, è sufficiente inserire il seguente codice all'interno del file `/js/theme/components/Button.js`: